client/src/main/java/org/eclipse/hono/client/ApplicationClientFactory.java

```diff
@@ -56,6 +56,29 @@ static ApplicationClientFactory create(final HonoConnection connection, final Se
      *
      * @param tenantId The tenant to consume data for.
      * @param telemetryConsumer The handler to invoke with every message received.
+     * @param autoAccept {@code true} if received deliveries should be automatically accepted (and settled)
```

Comment:
under which circumstances would you want to set this to false?

Reply:
In case accepting/rejecting the delivery should be done asynchronously.

In the case of the added test case here, a scenario should be simulated where the consumer takes a long time to accept the delivery - longer than the sendMessageTimeout. Without the `autoAccept=false` flag, this would mean having to block the event loop thread in the message handler, requiring the use of an extra HonoConnection/ApplicationClientFactory instance in the test.

Reply:
What you describe as the use is actually an anti pattern, isn't it? It means that the messages sent downstream will remain *in flight* for an indeterminate amount of time :-(
IMHO we should be very explicit in the JavaDoc that a consumer should not perform any time consuming work in the handler but should spawn a new thread or something like that in such a case, but otherwise quickly settle the incoming transfer.

Reply:
I've added a corresponding note in the javadocs.